src/components/ShareResultsModal/LinkField.tsx

```diff
@@ -10,6 +11,19 @@ interface LinkFieldProps {
 export default function LinkField({ url }: LinkFieldProps) {
   const [isCopied, setIsCopied] = useState(false);
   const [isError, setIsError] = useState(false);
+  const [showNotification, setShowNotification] = React.useState(false);
+
+  const onCloseToaster = () => {
+    setShowNotification(false);
+    setIsCopied(false);
+    setIsError(false);
+  };
+
+  useEffect(() => {
```

Comment:
Instead of using a `useEffect` here, would it be better to `setShowNotification` in the same place we set `isSubmitted` and `formError`?

Reply:
I get that it's simpler this way and more direct. But I like to think of the notification as a side-effect of submitting the form not as part of the `handleSubmit` logic itself. (Not applicable here) but say for example some other function than `handleSubmit` is updating `isSubmitted` or `formError`, we wouldn't worry about adding `setShowNotification` there as well. All we will need is the `useEffect` that run whenever  `isSubmitted` or `formError` despite of how they were changed